### PR TITLE
Added keyvault make recipes

### PR DESCRIPTION
**Pull request by enformatique11**
- Created the KV secret AZURE_CREDENTIALS and INFRASTRUCTURE for tra-trn-generation-api
- Updated the Makefile with azure_key_vault recipes to print, read, create and edit keyvault secret.
- Created a bin director to fectch_config.rb to create a secret via make recipe.
- Created a terraform directory with workspace_variables to fetch secrets, etc.

### CODE REVIEW

File: bin/fetch_config.rb [hunk 1]
```diff
@@ -0,0 +1,434 @@
+#!/usr/bin/env ruby
```
Comment (saliceti) [inline bin/fetch_config.rb:1]: This file should not be committed to the repository. Please add the path to .gitignore

Comment (enformatique11) [reply bin/fetch_config.rb:1]: done

File: .gitignore [hunk 2]
```diff
@@ -0,0 +1 @@
+fetch_config.rb
```
Comment (saliceti) [inline .gitignore:1]: It's not working as fetch_config.rb is still in this PR
Shouldn't it be bin/fetch_config.rb ?

File: bin/fetch_config.rb (hunk 1 above)
Comment (saliceti) [reply bin/fetch_config.rb:1]: It didn't work

File: .gitignore (hunk 2 above)
Comment (enformatique11) [reply .gitignore:1]: yep it should not be here

**Comment by saliceti** [comment]
I got this error:

```
% make dev print-keyvault-secret                                                                                                                                                    make: Nothing to be done for `dev'.
[ ! -f bin/fetch_config.rb ] \
		&& curl -s https://raw.githubusercontent.com/DFE-Digital/bat-platform-building-blocks/master/scripts/fetch_config/fetch_config.rb -o bin/fetch_config.rb \
		&& chmod +x bin/fetch_config.rb \
		|| true
echo "Logging on to s165-teachingqualificationsservice-development"
Logging on to s165-teachingqualificationsservice-development
az account set -s s165-teachingqualificationsservice-development
bin/fetch_config.rb -s azure-key-vault-secret:s165d01-trngen-dv-kv/INFRASTRUCTURE -f yaml
make: bin/fetch_config.rb: No such file or directory
make: *** [print-keyvault-secret] Error 1
```

**Review by saliceti — APPROVED** [review]

**Comment by enformatique11** [comment]
[Add secrets to 🔑 vault](https://trello.com/c/EgZppm8a/363-add-secrets-to-%F0%9F%94%91-vault)

**Pull request merged**